fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -358,20 +358,19 @@ public FDBRecordContext openContext(@Nullable Map<String, String> mdcContext,
                                         @Nullable FDBStoreTimer timer,
                                         @Nullable WeakReadSemantics weakReadSemantics,
                                         @Nonnull FDBTransactionPriority priority) {
-        final String transactionId = mdcContext == null ? null : mdcContext.get("uuid");
-        return openContext(mdcContext, timer, weakReadSemantics, priority, transactionId);
+        return openContext(mdcContext, timer, weakReadSemantics, priority, null);
     }
 
 
     /**
      * Open a new record context with a new transaction begun on the underlying FDB database.
      *
      * <p>
-     * Note that other variants of this method will inspect the MDC context for the transaction ID by looking
-     * for an entry in the map with the key "uuid". This method will ignore whatever is in the MDC context
-     * and use the ID provided as a parameter instead. The transaction ID should typically consist solely of
-     * printable ASCII characters and should not exceed 100 bytes. The ID may be truncated or dropped if the ID will
-     * not fit in 100 bytes. See {@link FDBRecordContext#getTransactionId()} for more details.
+     * If the passed {@code transactionId} is {@code null}, then this method will set the transaction ID
+     * of the given transaction to the value of the "uuid" key in the MDC context if present. The transaction ID
+     * should typically consist solely of printable ASCII characters and should not exceed 100 bytes. The ID may
+     * be truncated or dropped if the ID will not fit in 100 bytes. See {@link FDBRecordContext#getTransactionId()}
```

Comment:
This is alluding to that behavior change regarding the transaction ID that is mentioned in the PR's overview comment.